src/io/mod.rs

```diff
@@ -1,5 +1,113 @@
 //! Input and output of images.
+
+use crate::{error, ImageError, ImageResult};
+
 mod reader;
 pub(crate) mod free_functions;
 
 pub use self::reader::Reader;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+/// Set of supported strict limits for a decoder
+pub struct LimitSupport {
+    /// This field indicates whether the strict limit on the number of bytes is
+    /// supported.
+    pub max_alloc_strict: bool,
+    _non_exhaustive: (),
+}
+
+impl Default for LimitSupport {
+    fn default() -> LimitSupport {
+        LimitSupport {
+            max_alloc_strict: false,
+            _non_exhaustive: (),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
I know clippy will be annoying but I wouldn't derive `Copy` here, on purpose. Note that any future limit that requires an allocation space—consider a boxed callback—would be a breaking change otherwise.

Reply:
This seems like a good point